aptos-move/block-executor/src/executor.rs

```diff
@@ -655,7 +657,7 @@ where
                 base_view,
                 ViewState::Unsync(SequentialState {
                     unsync_map: &data_map,
-                    _counter: &0,
+                    counter: Rc::clone(&counter),
```

Comment:
If we are using Rc pointer here for SequentialState, shouldn't we be doing the same for ParallelState as well.

Reply:
sequential doesn't need to deal with threads, so doesn't need to be Atomic. Basically Rc<RefCell<>> is a looser non-thread safe variant, and Atomic is a stronger thread safe variant.

We could use Atomic here (with unnecessary overhead). we cannot use Rc<RefCell<>> in parallel execution